src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/SubjectAlternativeNameBuilder.cs

```diff
@@ -67,30 +69,31 @@ public void AddUserPrincipalName(string upn)
 
         public X509Extension Build(bool critical = false)
         {
-            AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+            _writer.Reset();
```

Comment:
Before for correctness, after so that we're eagerly clearing... which doesn't matter, since they're value types... so I should switch it to just before :)